src/services/GenerateVerifiableCredential.ts

```diff
@@ -0,0 +1,457 @@
+/* eslint-disable no-console */
+import { Logger } from "@aws-lambda-powertools/logger";
+import { AppError } from "../utils/AppError";
+import { HttpCodesEnum } from "../utils/HttpCodesEnum";
+import { YOTI_CHECKS, YotiSessionDocument } from "../utils/YotiPayloadEnums";
+import {
+	VerifiedCredentialEvidence,
+	VerifiedCredentialSubject,
+} from "../utils/IVeriCredential";
+
+export class GenerateVerifiableCredential {
+  readonly logger: Logger;
+
+  private static instance: GenerateVerifiableCredential;
+
+  constructor(logger: Logger) {
+  	this.logger = logger;
+  }
+
+  static getInstance(logger: Logger): GenerateVerifiableCredential {
+  	if (!GenerateVerifiableCredential.instance) {
+  		GenerateVerifiableCredential.instance = new GenerateVerifiableCredential(
+  			logger,
+  		);
+  	}
+  	return GenerateVerifiableCredential.instance;
+  }
+
+	/**
+		Checks if Document contains a chip and if that chip is valid but checking the chip_csca_trusted field
+	*/
+  private doesDocumentContainValidChip(documentType: string, documentAuthenticityCheckBreakdown: { sub_check: string; result: string; }[]): boolean {
+  	const validDocumentTypes = ["PASSPORT", "NATIONAL_ID", "RESIDENCE_PERMIT"];
```

Comment:
is this the right place to define this? the scope of this could potentially be broader than a check on whether the doc contains a valid chip...